tests/ed25519.rs

```diff
@@ -85,20 +84,21 @@ mod vectors {
     #[test]
     fn ed25519ph_rf8032_test_vector() {
         let secret_key: &[u8] = b"833fe62409237b9d62ec77587520911e9a759cec1d19755b7da901b96dca3d42";
-        let public_key: &[u8] = b"ec172b93ad5e563bf4932c70e1245034c35467ef2efd4d64ebf819683467e2bf";
+        let verifying_key: &[u8] =
+            b"ec172b93ad5e563bf4932c70e1245034c35467ef2efd4d64ebf819683467e2bf";
         let message: &[u8] = b"616263";
         let signature: &[u8] = b"98a70222f0b8121aa9d30f813d683f809e462b469c7ff87639499bb94e6dae4131f85042463c2a355a2003d062adf5aaa10b8c61e636062aaad11c2a26083406";
 
         let sec_bytes: Vec<u8> = FromHex::from_hex(secret_key).unwrap();
-        let pub_bytes: Vec<u8> = FromHex::from_hex(public_key).unwrap();
+        let pub_bytes: Vec<u8> = FromHex::from_hex(verifying_key).unwrap();
         let msg_bytes: Vec<u8> = FromHex::from_hex(message).unwrap();
         let sig_bytes: Vec<u8> = FromHex::from_hex(signature).unwrap();
 
-        let secret: SecretKey = SecretKey::from_bytes(&sec_bytes[..SECRET_KEY_LENGTH]).unwrap();
-        let expected_public: PublicKey =
-            PublicKey::from_bytes(&pub_bytes[..PUBLIC_KEY_LENGTH]).unwrap();
-        let keypair: Keypair = Keypair::from(secret);
-        assert_eq!(expected_public, keypair.public_key());
+        let signing_key: SigningKey =
+            SigningKey::try_from(&sec_bytes[..SECRET_KEY_LENGTH]).unwrap();
```

Comment:
Why does this work? Aren't `SecretKey` and `SigningKey` different?

Reply:
They're different: `SecretKey` is just a `[u8; 32]` alias. That said, I'm not sure why it's relevant here?

I switched `SigningKey` to use the 32-byte RFC8032 serialization of what it calls a "private key".

The 64-byte serialization which includes the public key as well is preserved as `from_keypair_bytes` and `to_keypair_bytes`, however I think we should discourage those as they are a nonstandard serialization based on effectively transmuting the ref10 key type, because SUPERCOP didn't need or care about serialization APIs. They only make the key longer and leave a sharp edge for the public key oracle.